pkg/catalog/manager/traverse.go

```diff
@@ -20,7 +20,36 @@ import (
 	"k8s.io/apimachinery/pkg/labels"
 )
 
+func (m *Manager) dropDeprecatedFields(cmt *CatalogInfo, catalogType string) error {
+	switch catalogType {
+	case client.CatalogType:
+		catalog := cmt.catalog
+		if catalog.Status.HelmVersionCommits == nil {
+			return nil
+		}
+		catalog.Status.HelmVersionCommits = nil
+	case client.ClusterCatalogType:
+		clusterCatalog := cmt.clusterCatalog
+		if clusterCatalog.Status.HelmVersionCommits == nil {
+			return nil
+		}
+		clusterCatalog.Status.HelmVersionCommits = nil
+	case client.ProjectCatalogType:
+		projectCatalog := cmt.projectCatalog
+		if projectCatalog.Status.HelmVersionCommits == nil {
+			return nil
+		}
+		projectCatalog.Status.HelmVersionCommits = nil
+	}
+	_, err := m.updateCatalogInfo(cmt, catalogType, "", true, true)
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
This can be simplified to `return err`, but might want to be kept for readability. Leaving this open for reviewers' thoughts here.

Reply:
Decided to make the optimization